src/python/python-kdmp.cc

```diff
@@ -0,0 +1,277 @@
+
+    //
+    // Parse Python argument (expect one unsigned long long integer).
+    //
+
+    PythonDumpParser *Self = reinterpret_cast<PythonDumpParser*>(Object);
+
+    uint64_t PhysicalAddress = 0;
+    if (!PyArg_ParseTuple(Args, "K", &PhysicalAddress)) {
+        return PyErr_Format(PyExc_TypeError, "get_physical_page() expected an integer");
+    }
+
+    //
+    // Get the physical page and return it as bytes.
+    //
+
+    const uint8_t *Page = Self->DumpParser->GetPhysicalPage(PhysicalAddress);
+
+    if (!Page) {
+        return PyErr_Format(PyExc_ValueError, "get_physical_page() invalid address");
+    }
+
+    return PyBytes_FromStringAndSize(reinterpret_cast<const char*>(Page), Page::Size);
+}
+
+//
+//  Python Dump instance method to perform address translation (physical to virtual).
+//  >>> dump_instance.virt_translate(addr, [dtb]) # return int
+//
+
+PyObject* DumpParserVirtTranslate(PyObject *Object, PyObject *Args) {
+
+    //
+    // Parse Python argument (expect one or two unsigned long long integer).
+    //
+
+    PythonDumpParser *Self = reinterpret_cast<PythonDumpParser*>(Object);
+
+    uint64_t VirtualAddress = 0;
+    uint64_t DirectoryTableBase = 0;
+    if (!PyArg_ParseTuple(Args, "K|K", &VirtualAddress, &DirectoryTableBase)) {
+        return PyErr_Format(PyExc_TypeError, "translate_address() expected one or two integer");
+    }
+
+    //
+    // Retrieve the physical address (parse pages tables in the dump).
+    //
+
+    const uint64_t PhysicalAddress = Self->DumpParser->VirtTranslate(VirtualAddress, DirectoryTableBase);
+
+    if (!PhysicalAddress) {
+        return PyErr_Format(PyExc_ValueError, "translate_address() invalid address");
+    }
+
+    return PyLong_FromUnsignedLongLong(PhysicalAddress);
+}
+
+//
+//  Python Dump instance method to get a page from a virtual address.
+//  >>> dump_instance.get_virtual_page(addr, [dtb]) # return bytes
+//
+
+PyObject* DumpParserGetVirtualPage(PyObject *Object, PyObject *Args) {
+
+    //
+    // Parse Python argument (expect one or two unsigned long long integer).
+    //
+
+    PythonDumpParser *Self = reinterpret_cast<PythonDumpParser*>(Object);
+
+    uint64_t VirtualAddress = 0;
+    uint64_t DirectoryTableBase = 0;
+    if (!PyArg_ParseTuple(Args, "K|K", &VirtualAddress, &DirectoryTableBase)) {
+        return PyErr_Format(PyExc_TypeError, "get_virtual_page() expected one or two integer");
+    }
+
+    const uint8_t *Page = Self->DumpParser->GetVirtualPage(VirtualAddress, DirectoryTableBase);
+
+    if (!Page) {
+        return PyErr_Format(PyExc_ValueError, "get_virtual_page() invalid address");
+    }
+
+    return PyBytes_FromStringAndSize(reinterpret_cast<const char*>(Page), Page::Size);
+}
+
+//
+// KDMP Module initialization function.
+//
+
+PyMODINIT_FUNC
+PyInit_kdmp(void)
+{
+
+    //
+    // Initialize python.
+    //
+
+    Py_Initialize();
+
+    //
+    // Register PythonDumpParserType.
+    //
+
+	if(PyType_Ready(&PythonDumpParserType) < 0)

```

Comment:
The indentation doesn't look right here - I can run clang-format once merged if you want me to

Reply:
yes please,  I will resolve the last comments and rebase the commit first.